packages/ui/core-components/src/lib/molecules/dimension-grid/DimensionRow.svelte

```diff
@@ -23,14 +23,24 @@
 	<div
 		class={cn(
 			'bg-blue-50 group-hover:bg-blue-100 absolute inset-y-0 left-0 z-[-10]',
-			value.includes('NaN') ? 'bg-gray-200' : isSelected ? 'bg-blue-200' : ''
+			{
+				// undefined occurs in multi-selects where the user has selected mutually exclusive options (see null row column combination story)
+				'bg-gray-200': row.metric === undefined
+			},
+			{
+				// null can occur naturally, allow selection state to show through
+				'bg-transparent': row.metric === null
+			},
+			{
+				'bg-blue-200': row.metric && isSelected
+			}
 		)}
```

Comment:
Could this be written with a series of `class:bg-someclass={condition}` rather than this object syntax? I'm not sure we use it elsewhere and introducing a new way to do conditional classes seems like a big step